python/pyproject.toml

```diff
@@ -35,7 +35,7 @@ dependencies = [
 [project.optional-dependencies]
 docs = ["markdown", "pyyaml", "sphinx", "sphinx_rtd_theme"]
 lint = ["ruff"]
-optional = ["numba"]
+optional = ["numba", "pyamg"]
```

Comment:
`pyamg` shouldn't be added here, it's require for a demo but is not used anywhere in the core.